### PR TITLE
Add events order + fixes to behavior doc 

### PR DESCRIPTION
**Pull request by blikblum**
### Proposed changes
 - Document order of events between behavior and view
 - Merge Using ui and referencing DOM since the subject is the same
 - Add note about triggering an event in the view

The example is created in an anonymous fiddle. It should be migrated to an marionettejs account?

### CODE REVIEW

File: docs/marionette.behavior.md [hunk 1]
```diff
@@ -16,8 +16,10 @@ collection interactions to be utilized across your application.
 * [View Proxy](#view-proxy)
   * [Listening to View Events](#listening-to-view-events)
   * [Proxy Handlers](#proxy-handlers)
+  * [Events Order](#events-order)
+  * [Using ui](#using-ui)
```
Comment (scott-w) [inline docs/marionette.behavior.md]: Can you make this: 
```
  * [Using `ui`](#using-ui)
```

File: docs/marionette.behavior.md [hunk 2]
```diff
@@ -316,10 +321,23 @@ including:
...
+
+If both view and behavior are listening for the same event, this will be executed
+first in the view than in the behavior. 
+
+Unlike events, the initialize method will 
+be called first in the behavior. 
+  
+> While initialize is called in behavior before than in view, the view property is already set, and can be accessed, inside behavior initialize 
+  
+[Live example](https://jsfiddle.net/9wtqgwgu/)
```
Comment (denar90) [inline docs/marionette.behavior.md]: It's better to use `marionettejs` account for fiddle.

File: docs/marionette.behavior.md (hunk 1 above)
Comment (blikblum) [inline docs/marionette.behavior.md]: Done

File: docs/marionette.behavior.md (hunk 2 above)
Comment (blikblum) [inline docs/marionette.behavior.md]: How it can be done?

Comment (blikblum) [inline docs/marionette.behavior.md]: While initialize is called in behavior before than in view, the view property is already set, and can be accessed, inside behavior initialize 

The above sentence, is not very good. I tried various versions. If someone has suggestions, please do.

Comment (scott-w) [inline docs/marionette.behavior.md]: I usually do a heading of sorts - something like:

```markdown
#### How Behavior is Initialized

The View+Behavior initialize process is as follows:

1. View is constructed
2. Behavior is constructed
3. Behavior is initialized with attached view
4. View is initialized

This means that the Behavior can see an unitialized view during its own initialize setup and can inject information into the view to consider for its own initialize steps.
```

Comment (denar90) [inline docs/marionette.behavior.md]: As addition to @scott-w comment some info can be grab from https://github.com/marionettejs/backbone.marionette/issues/1579

Comment (blikblum) [inline docs/marionette.behavior.md]: @denar90 , thanks. #1579 was one of my source of information although the initial order is not valid anymore. I updated the fiddle to show constructor: https://jsfiddle.net/marionettejs/qb9go1y3/1/ .

This knowledge was one that i missed and had to test myself when i was learning how to use behavior. As the other changes i'm proposing, it is inspired by the lacks i observed reading the docs.

File: docs/marionette.behavior.md [hunk 3]
```diff
@@ -16,8 +16,10 @@ collection interactions to be utilized across your application.
 * [View Proxy](#view-proxy)
   * [Listening to View Events](#listening-to-view-events)
   * [Proxy Handlers](#proxy-handlers)
+  * [Events / Initialize Order](#events-initialize-order)
```
Comment (paulfalgout) [inline docs/marionette.behavior.md]: This is going to be `#events--initialize-order`

File: docs/marionette.behavior.md [hunk 4]
```diff
@@ -306,6 +308,9 @@ triggered on a `View` are passed to all attached `behaviors`. This includes:
 These handlers work exactly as they do on `View` -
 [see the `View` documentation](./marionette.view.md#events)
 
+> Events triggered in behavior instance are not executed in view. To notify
```
Comment (paulfalgout) [inline docs/marionette.behavior.md]: can we say "in the behavior instance" and "in the view" to match the next sentence?

File: docs/marionette.behavior.md [hunk 5]
```diff
@@ -316,10 +321,28 @@ including:
 * [`modelEvents`](./marionette.view.md#model-events)
 * [`collectionEvents`](./marionette.view.md#collection-events)
 
+### Events / Initialize Order
+
+If both view and behavior are listening for the same event, this will be executed
+first in the view than in the behavior as below.
```
Comment (paulfalgout) [inline docs/marionette.behavior.md]:
```diff
- than
+ then
```

File: docs/marionette.behavior.md (hunk 4 above)
Comment (blikblum) [inline docs/marionette.behavior.md]: Done

File: docs/marionette.behavior.md (hunk 3 above)
Comment (blikblum) [inline docs/marionette.behavior.md]: done

**Comment by paulfalgout** [comment]
Just noticed this introduced doc linter errors: https://travis-ci.org/marionettejs/backbone.marionette/builds/173784807#L987

@denar90 we should probably make a doc linter error fail the build.

**Comment by denar90** [comment]
@paulfalgout these are warning. But yeah it's better to fail build so we can avoid lint warnings in future.

**Comment by blikblum** [comment]
I will open a new pr fixing them

**Pull request merged**